src/ILLink.Tasks/CreateRuntimeRootDescriptorFile.cs

```diff
@@ -54,9 +54,33 @@ class ClassMembers
 			public HashSet<string> fields;
 		}
 
+		/// <summary>
+		/// Helper utility to track feature switch macros in header file
+		/// This type is used in a dictionary as a key and only uses
+		/// the feature name since value and default are not expected to change
```

Comment:
> the feature name since value and default are not expected to change

I think this is a dangerous assumption to make. Please change this so that all three values are part of the key.